### PR TITLE
bumped ruby version to 3 and faraday version to 2

### PR DESCRIPTION
**Pull request by obroshnij**
This PR adds support for ruby v.3.2 and faraday version 2 (https://github.com/lostisland/faraday/discussions/1358) which is shipped with a bunch of optimizations and improvements out of the box

### CODE REVIEW

File: test/lib/adapter_test.rb [hunk 1]
```diff
@@ -17,8 +28,8 @@ def test_default
         net_http_persistent: Faraday::Adapter::NetHttpPersistent,
         typhoeus: Faraday::Adapter::Typhoeus,
         patron: Faraday::Adapter::Patron,
-        em_synchrony: Faraday::Adapter::EMSynchrony,
-        em_http: Faraday::Adapter::EMHttp,
+        # em_synchrony: Faraday::Adapter::EMSynchrony,
+        # em_http: Faraday::Adapter::EMHttp,
```
Comment (UmanShahzad) [inline test/lib/adapter_test.rb:32]: Can we just rm?

File: test/lib/adapter_test.rb [hunk 2]
```diff
@@ -3,6 +3,17 @@
...
+require 'faraday/net_http_persistent'
+require 'faraday/typhoeus'
+require 'faraday/patron'
+require 'faraday/httpclient'
+require 'faraday/excon'
+
+# No updates since 2021, the method `dependecy` from the class Faraday::Adapter has been removed
+# https://github.com/lostisland/faraday/blob/main/UPGRADING.md#the-dependency-method-in-middlewares-has-been-removed
+# require 'faraday/em_http'
+# require 'faraday/em_synchrony'
```
Comment (UmanShahzad) [inline test/lib/adapter_test.rb:15]: Can we just rm?

**Review by UmanShahzad — APPROVED** [review]

File: test/lib/adapter_test.rb (hunk 2 above)
Comment (UmanShahzad) [inline test/lib/adapter_test.rb:15]:
```suggestion
```

File: test/lib/adapter_test.rb (hunk 1 above)
Comment (UmanShahzad) [inline test/lib/adapter_test.rb:32]:
```suggestion
```

**Review by UmanShahzad — APPROVED** [review]

**Comment by obroshnij** [comment]
thanks @UmanShahzad

**Pull request merged**